### PR TITLE
Create 2018-07-11-dynamic-kubelet-configuration.md

### PR DESCRIPTION
**Pull request by Sophy417**

### CODE REVIEW

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+-->
+Kubernetes 提供了以 API 为中心的工具，可显着改善用于管理应用程序和基础架构的工作流程。
+但是，大多数 Kubernetes 安装都在标准 Kubernetes API 范围之外，在每个主机上将 Kubelet 作为本机进程运行。
+
+<!--
+In the past, this meant that cluster administrators and service providers could not rely on Kubernetes APIs to reconfigure Kubelets in a live cluster. In practice, this required operators to either ssh into machines to perform manual reconfigurations, use third-party configuration management automation tools, or create new VMs with the desired configuration already installed, then migrate work to the new machines. These approaches are environment-specific and can be expensive.
+-->
+过去，这意味着集群管理员和服务提供商无法依靠 Kubernetes API 在活动集群中重新配置 Kubelets。
+实际上，这要求操作员要么转入计算机以执行手动重新配置，要么使用第三方配置管理自动化工具，或创建已经安装了所需配置的新 VM，然后将工作迁移到新计算机上。
+这些方法是特定于环境的，并且可能很昂贵。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:40]:
```suggestion
这些方法是特定于环境的，并且可能很耗时费力。
```

expensive 在这里跟价格无关

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+<!--
+## What is Dynamic Kubelet Configuration?
+-->
+## 什么是动态 Kubelet 配置？
+
+<!--
+Kubernetes v1.10 made it possible to configure the Kubelet via a beta [config file](/docs/tasks/administer-cluster/kubelet-config-file/) API. Kubernetes already provides the ConfigMap abstraction for storing arbitrary file data in the API server.
+-->
+Kubernetes v1.10 使得可以通过beta [配置文件](/docs/tasks/administer-cluster/kubelet-config-file/)API 配置 Kubelet。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:55]:
```suggestion
Kubernetes v1.10 使得可以通过 Beta 版本的[配置文件](/docs/tasks/administer-cluster/kubelet-config-file/)
API配置 kubelet。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+Kubernetes 已经提供了 ConfigMap 抽象，用于在 API 服务器中存储任意文件数据。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:56]:
```suggestion
Kubernetes 已经提供了用于在 API 服务器中存储任意文件数据的 ConfigMap 抽象。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+动态 Kubelet 配置提供以下核心功能：
+
+<!--
+* Kubelet attempts to use the dynamically assigned configuration.
+* Kubelet "checkpoints" configuration to local disk, enabling restarts without API server access.
+* Kubelet reports assigned, active, and last-known-good configuration sources in the Node status.
+* When invalid configuration is dynamically assigned, Kubelet automatically falls back to a last-known-good configuration and reports errors in the Node status.
+-->
+* Kubelet 尝试使用动态分配的配置。
+* Kubelet 对本地磁盘的“检查点”配置，无需 API 服务器访问即可重新启动。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:81]:
```suggestion
* Kubelet 将其配置已检查点的形式保存到本地磁盘，无需 API 服务器访问即可重新启动。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+* Kubelet 报告处于“节点”状态的已分配，活动和最近已知良好的配置源。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:82]:
```suggestion
* Kubelet 在 Node 状态中报告已指定的、活跃的和最近已知良好的配置源。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+* 当动态分配了无效的配置时，Kubelet 会自动退回到最后一次正确的配置，并报告节点状态中的错误。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:83]:
```suggestion
* 当动态分配了无效的配置时，Kubelet 会自动退回到最后一次正确的配置，并在 Node 状态中报告错误。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+
+<!--
+To use the dynamic Kubelet configuration feature, a cluster administrator or service provider will first post a ConfigMap containing the desired configuration, then set each Node.Spec.ConfigSource.ConfigMap reference to refer to the new ConfigMap. Operators can update these references at their preferred rate, giving them the ability to perform controlled rollouts of new configurations.
+-->
+要使用动态 Kubelet 配置功能，群集管理员或服务提供商将首先发布包含所需配置的 ConfigMap，然后设置每个 Node.Spec.ConfigSource。
+ConfigMap 参考以引用新的 ConfigMap。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:89]:
```suggestion
要使用动态 Kubelet 配置功能，群集管理员或服务提供商将首先发布包含所需配置的 ConfigMap，
然后设置每个 Node.Spec.ConfigSource.ConfigMap 引用以指向新的 ConfigMap。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+运营商可以以他们喜欢的速率更新这些参考，从而使他们能够执行新配置的受控部署。
+
+<!--
+Each Kubelet watches its associated Node object for changes. When the Node.Spec.ConfigSource.ConfigMap reference is updated, the Kubelet will "checkpoint" the new ConfigMap by writing the files it contains to local disk. The Kubelet will then exit, and the OS-level process manager will restart it. Note that if the Node.Spec.ConfigSource.ConfigMap reference is not set, the Kubelet uses the set of flags and config files local to the machine it is running on.
+-->
+每个 Kubelet 都会监视其关联的 Node 对象的更改。
+更新 Node.Spec.ConfigSource.ConfigMap 引用后，
+Kubelet 将通过将其包含的文件写入本地磁盘来“检查点”新的 ConfigMap。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:97]:
```suggestion
Kubelet 将通过将其包含的文件通过检查点机制写入本地磁盘保存新的 ConfigMap。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 9]
```diff
@@ -0,0 +1,132 @@
...
+然后，Kubelet 将退出，并且操作系统级进程管理者将重新启动它。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:98]:
```suggestion
然后，Kubelet 将退出，而操作系统级进程管理器将重新启动它。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+请注意，如果未设置 Node.Spec.ConfigSource.ConfigMap 引用，
+则 Kubelet 将使用其正在运行的计算机本地的一组标志和配置文件。
+
+<!--
+Once restarted, the Kubelet will attempt to use the configuration from the new checkpoint. If the new configuration passes the Kubelet's internal validation, the Kubelet will update Node.Status.Config to reflect that it is using the new configuration. If the new configuration is invalid, the Kubelet will fall back to its last-known-good configuration and report an error in Node.Status.Config.
+-->
+重新启动后，Kubelet 将尝试从新检查点使用配置。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:105]:
```suggestion
重新启动后，Kubelet 将尝试使用来自新检查点的配置。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 11]
```diff
@@ -0,0 +1,132 @@
...
+如果新配置通过了 Kubelet 的内部验证，则 Kubelet 将更新 Node。
+Status.Config 用以反映它正在使用新配置。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:107]:
```suggestion
如果新配置通过了 Kubelet 的内部验证，则 Kubelet 将更新 Node.Status.Config 用以反映它正在使用新配置。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 12]
```diff
@@ -0,0 +1,132 @@
...
+
+<!--
+## How can I learn more?
+-->
+## 我如何了解更多？
+
+<!--
+Please see the official tutorial at /docs/tasks/administer-cluster/reconfigure-kubelet/, which contains more in-depth details on user workflow, how a configuration becomes "last-known-good," how the Kubelet "checkpoints" config, and possible failure modes.
+-->
+请参阅/ docs / tasks / administer-cluster / reconfigure-kubelet /上的官方教程，
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:131]: 这些空格怎么来的？

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 13]
```diff
@@ -0,0 +1,132 @@
...
+其中包含有关用户工作流，配置如何成为“最新的正确的”配置，Kubelet如何“检查点”配置的更多详细信息，以及可能的故障模式。
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:132]:
```suggestion
其中包含有关用户工作流，某配置如何成为“最新的正确的”配置，Kubelet如何对配置执行“检查点”操作等
更多详细信息，以及可能的故障模式。
```

File: content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md [hunk 14]
```diff
@@ -52,8 +54,9 @@ Dynamic Kubelet configuration gives cluster administrators and service providers
 <!--
 Kubernetes v1.10 made it possible to configure the Kubelet via a beta [config file](/docs/tasks/administer-cluster/kubelet-config-file/) API. Kubernetes already provides the ConfigMap abstraction for storing arbitrary file data in the API server.
 -->
-Kubernetes v1.10 使得可以通过beta [配置文件](/docs/tasks/administer-cluster/kubelet-config-file/)API 配置 Kubelet。
-Kubernetes 已经提供了 ConfigMap 抽象，用于在 API 服务器中存储任意文件数据。
+Kubernetes v1.10 使得可以通过 Beta 版本的[配置文件](/docs/tasks/administer-cluster/kubelet-config-file/)
```
Comment (tengqm) [inline content/zh/blog/_posts/2018-07-11-dynamic-kubelet-configuration.md:57]:
```suggestion
Kubernetes v1.10 使得可以通过 Beta 版本的[配置文件](/zh/docs/tasks/administer-cluster/kubelet-config-file/)
```

**Comment by tengqm** [comment]
If you do not know how to squash commits, better learn how to do it.

**Comment by Sophy417** [comment]
> If you do not know how to squash commits, better learn how to do it.

好的，我之后会学习的，不好意思，这几天翻译的任务比较多，比较忙。请问我前面几个已经纠错过的翻译什么时候能合并进去？

**Pull request merged**